Restructure Build apps: add an Arbitrum essentials section (#3332)
OffchainLabs/arbitrum-docs: docs/arbitrum-essentials/arbitrum-vs-ethereum/comparison-overview.mdx

### Block numbers and time

> Time in Arbitrum chains is tricky. The timing assumptions one is used to making about Ethereum blocks don't exactly carry over into the timing of Arbitrum blocks. See [Block numbers and time](/arbitrum-essentials/arbitrum-vs-ethereum/block-numbers-and-time.mdx) for details about how block numbers and time work in Arbitrum.

Author: ```suggestion
Time in Arbitrum chains is tricky. The timing assumptions that apply to Ethereum blocks don't exactly carry over to Arbitrum blocks. See [Block numbers and time](/arbitrum-essentials/arbitrum-vs-ethereum/block-numbers-and-time.mdx) for details about how block numbers and time work in Arbitrum.
```

### RPC methods

> Although the majority of RPC methods follow the same behavior as Ethereum, some methods may produce a different result or add additional information when used on an Arbitrum chain. You can find more information about these differences in [RPC methods](/arbitrum-essentials/arbitrum-vs-ethereum/rpc-methods.mdx).

Author: ```suggestion
Although the majority of RPC methods follow the same behavior as Ethereum, some methods may produce a different result or add additional information when used on an Arbitrum chain. For more details, see [RPC methods](/arbitrum-essentials/arbitrum-vs-ethereum/rpc-methods.mdx).
```

### Precompiles

> Besides supporting all precompiles available in Ethereum, Arbitrum provides child chain-specific precompiles with methods that smart contracts can call in the same way as Solidity functions. You can find a full reference to them on the [Precompiles](/arbitrum-essentials/precompiles/overview.mdx) page.

Author: ```suggestion
Besides supporting all precompiles available in Ethereum, Arbitrum provides child chain-specific precompiles with methods that smart contracts can call in the same way as Solidity functions. For a full reference, see the [Precompiles](/arbitrum-essentials/precompiles/overview.mdx) page.
```

### NodeInterface

> The Arbitrum Nitro software includes a special `NodeInterface` contract, available at address `0xc8`, that is only accessible via RPCs (deployed offchain, making it inaccessible to smart contracts). Find more information about this interface in [NodeInterface](/arbitrum-essentials/nodeinterface/overview.mdx).

Author: ```suggestion
The Arbitrum Nitro software includes a special `NodeInterface` contract, available at address `0xc8`, that is only accessible via RPCs (deployed offchain, making it inaccessible to smart contracts). For more details, see [NodeInterface](/arbitrum-essentials/nodeinterface/overview.mdx).
```
